### PR TITLE
Client Side API Creation of Test Settings

### PR DESCRIPTION
**Pull request by Madhurima94**
Files Changed-tests.routing.ts,
 tests-dashboard.html,
test-settings.html,
test-settings.component,
tests.module.ts,
tests.model.ts
TestSettingsRepository.cs
ITestSettingsRepository.cs
TestSettingsController.cs
Files Created- testsettings.service.ts
Tasks Created-#20890,#20891,#20903,#21031,#21032,#21033,#21052

### CODE REVIEW

File: Trappist/src/Promact.Trappist.Repository/TestSettings/TestSettingsRepository.cs [hunk 1]
```diff
@@ -24,22 +23,32 @@ public TestSettingsRepository(TrappistDbContext dbContext)
...
         /// </summary>
-        /// <returns>Settings set for that Test</returns>
-        public List<Test> GetTestSettings()
+        /// <param name="id">The parameter "id" is used to get the Settings of a Test by its Id</param>
+        /// <returns>Settings Saved for the selected Test</returns>
+        public Test GetTestSettings(int id)
         {
-            var settings = _dbContext.Test.ToList();
-            return settings;
+            var get = _dbContext.Test.FirstOrDefault(x => x.Id == id);
```
Comment (DhrutiShah) [inline Trappist/src/Promact.Trappist.Repository/TestSettings/TestSettingsRepository.cs]: variable name should be based on details which are rutuning from query. Please change

File: Trappist/src/Promact.Trappist.Repository/TestSettings/TestSettingsRepository.cs [hunk 2]
```diff
@@ -24,22 +23,32 @@ public TestSettingsRepository(TrappistDbContext dbContext)
...
+            return get;
```
Comment (DhrutiShah) [inline Trappist/src/Promact.Trappist.Repository/TestSettings/TestSettingsRepository.cs]: variable name is not proper

File: Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-multiple-answers/questions-multiple-answers.html [hunk 3]
```diff
@@ -2,7 +2,7 @@
     <div class="question-type">
         <div class="header">
             <h2>Question</h2>
-            <h5>Multiple Answer Question</h5>
+
```
Comment (DhrutiShah) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-multiple-answers/questions-multiple-answers.html]: Make sure , it is looks like other's code. but included in your commit.

File: Trappist/src/Promact.Trappist.Web/wwwroot/app/tests/test-settings/test-launch-dialog.html [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
+            <label class="h4">Global URL</label>
+            <div class="url-container">
+                <input type="text" class="form-control" [(ngModel)]="settingObject.link" name="Test Link"/>
+                <i class="material-icons" ngxClipboard [cbContent]="settingObject.link" (cbOnSuccess)="isCopied1 = true">content_copy</i>
+            </div>
+        </div>
+    </div>
+    <div class="dialog-actions">
+        <button type="button" class="btn" md-dialog-close>Close</button>
+        <button type="submit" class="btn btn-primary" [class.btn-success]="isCopied1" ngxClipboard [cbContent]="settingObject.link" (cbOnSuccess)="isCopied1 = true">Copy</button>
```
Comment (DhrutiShah) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/tests/test-settings/test-launch-dialog.html]: take proper variable name.

File: Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-multiple-answers/questions-multiple-answers.html (hunk 3 above)
Comment (Madhurima94) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-multiple-answers/questions-multiple-answers.html]: I will take care next time.

**Pull request merged**